include/swift/AST/DiagnosticsIRGen.def

```diff
@@ -59,5 +59,10 @@ ERROR(temporary_allocation_alignment_not_positive,none,
 ERROR(temporary_allocation_alignment_not_power_of_2,none,
       "alignment value must be a power of two", ())
 
+ERROR(foreign_reference_types_unsupported,none,
```

Comment:
@rjmccall and @beccadax I need to add a test for this new error, but other than that, this should be ready for re-review.